plugins/modules/dcnm_service_node.py

```diff
@@ -0,0 +1,741 @@
+  config:
+    description: 'List of details of service nodes being managed'
+    type: list
+    elements: dict
+    required: true
+    note: Not required for state deleted
+    suboptions:
+      name:
+        description: 'Name of service node'
+        type: str
+        required: true
+      type:
+        description: 'Name of the service node type'
+        type: str
+        required: true
+        default: 'firewall'
+      form_factor:
+        description: 'Name of the form factor of the service node'
+        type: str
+        required: true
+        default: 'physical'
+      svc_int_name:
+        description: 'Name of the service interface'
+        type: str
+        required: true
+      switches:
+        description: 'IP address of the switch where service node will be added/deleted'
+        type: list
+        required: true
+      attach_interface:
+        description: 'List of switch interfaces where the service node will be attached'
+        type: str
+        required: true
+'''
+
+EXAMPLES = '''
+This module supports the following states:
+
+Merged:
+  Service Nodes defined in the playbook will be merged into the service fabric.
+    - If the service node does not exist it will be added.
+    - If the service node exists but properties managed by the playbook are different
+      they will be updated if possible.
+    - Service Nodes that are not specified in the playbook will be untouched.
+
+Replaced:
+  Service Nodes defined in the playbook will be replaced in the service fabric.
+    - If the service node does not exist it will not be added.
+    - If the service node exists but properties managed by the playbook are different
+      they will be updated if possible.
+    - Properties that can be managed by the module but are not specified
+      in the playbook will be deleted or defaulted if possible.
+    - Service node that are not specified in the playbook will be untouched.
+
+Overridden:
+  Service Node defined in the playbook will be overridden in the service fabric.
+    - If the service node does not exist it will be added.
+    - If the service node exists but properties managed by the playbook are different
+      they will be updated if possible.
+    - Properties that can be managed by the module but are not specified
+      in the playbook will be deleted or defaulted if possible.
+    - Service Node that are not specified in the playbook will be deleted.
+
+Deleted:
+  Service Node defined in the playbook will be deleted.
+  If no Service Node are provided in the playbook, all service node present on that DCNM fabric will be deleted.
+
+Query:
+  Returns the current DCNM state for the service node listed in the playbook.
+
+- name: Merge Service Nodes
+  cisco.dcnm.dcnm_service_node:
+    fabric: Fabric1
+    service_fabric: external
+    state: merged 
+    config:
+    - name: SN-11
+      type: firewall
+      form_factor: virtual
+      svc_int_name: svc1
+      attach_interface: Ethernet1/1
```

Comment:
The documentation states that `attach_interface` can be a list of interfaces. Have we tested multiple interfaces?

Reply:
Yes Mike, for single device single interface is supported and it is tested, for switches upto 2 vpc is the only interface option and even that is tested extensively.